misc/h2olog/README.md

```diff
@@ -0,0 +1,83 @@
+# h2olog
```

Comment:
I think we might want to move this to `srcdoc` and generate HTML & man from there?